Source/Dafny/Compiler-java.cs

```diff
@@ -3483,20 +3483,24 @@ private class GenericArrayElementLvalue : ILvalue {
       }
     }
 
-    private void TrExprAsInt(Expression expr, TargetWriter wr, bool inLetExprBody) {
+    private void TrExprAsInt(Expression expr, TargetWriter wr, bool inLetExprBody, bool checkRange = false, string msg = null) {
```

Comment:
Are `checkRange` and `msg` independent parameters? If not, please write a contract about them. For example:

    Contract.Requires(!checkRange || msg != null);

If the relation between them turns out to be `checkRange == (msg != null)`, then I suggest eliminating the `checkRange` parameter and adding a descriptive comment for the method that explains that passing in `msg` as non-null gives rise to a checked conversion.

Reply:
They are independent, except that if checkRange == false then msg is ignored.
Code cleaned up and commented to indicate this